### ISSUE TITLE
Enforce code formatting

### ISSUE THREAD

**Issue by gayanW** [javapathfinder/jpf-core#160]
The formatting of source code is repeatedly inconsistent. (Eg. both tabs and spaces being used for indenting even within the same file)

What if we enforce formatting to the Gradle build. So, for instance, we could add a task that reformats source code to comply with [Google Java Style](https://google.github.io/styleguide/javaguide.html). I've tested this with [spotless](https://github.com/diffplug/spotless) plugin and it works great.

**Comment by cyrille-artho** [comment]
The problem is that reformatting initially introduces a huge patch set. Therefore, it would be best to reformat the code after we have achieved a particular milestone, such as Java 10 compatibility, before merging the code to master.
Reformatting would then introduce one giant commit that changes most files, but after that, files would be mostly untouched.
I'm not sure if reformatting as part of the build is a good idea, as it will mess up with undo/redo of files being edited at that time. It would probably be best to have a git commit hook for that task instead.

**Comment by gayanW** [comment]
Yeah, it is true that the initial patch is large. Several open source projects seem to use such plugins, so to make sure that their formatting remains consistent.

spotless tasks such as `spotlessJavaCheck`, `spotlessApply` also can be run manually, so I'm not sure whether that it would affect undo/redo the process. Let's keep this for later :)
